akka-remote/src/main/scala/akka/remote/internal/Hash128.scala

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (C) 2023 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.remote.internal
+
+import java.lang.{ Long => JLong }
+
+import scala.annotation.switch
+import scala.annotation.tailrec
+
+import akka.annotation.InternalApi
+
+/**
+ * INTERNAL API
+ *
+ * This contains a Scala port of the 128-bit hash function `MurmurHash3_x64_128` from Austin Appleby's original
+ * `c++` code in SMHasher. It was ported from `MurmurHash3` in commons-codec,
+ *
+ * SMHasher public domain code with no copyrights. Commons codec is licensed under the Apache License, Version 2.0.
+ */
+@InternalApi private[akka] object Hash128 {
+
+  private val C1 = 0x87C37B91114253D5L
+  private val C2 = 0x4CF5AD432745937FL
+  private val R1 = 31
+  private val R2 = 27
+  private val R3 = 33
+  private val M = 5
+  private val N1 = 0x52dce729
+  private val N2 = 0x38495ab5
+
+  /**
+   * Generates 128-bit hash from the byte array.
+   *
+   * This is an implementation of the 128-bit hash function `MurmurHash3_x64_128`
+   * from Austin Appleby's original MurmurHash3 `c++` code in SMHasher.
+   */
+  def hash128x64(data: Array[Byte]): (Long, Long) =
+    hash128x64(data, 0, data.length, 0)
+
+  /**
+   * Generates 128-bit hash from the byte array with the given offset, length and seed.
+   *
+   * This is an implementation of the 128-bit hash function `MurmurHash3_x64_128`
+   * from Austin Appleby's original MurmurHash3 `c++` code in SMHasher.
+   */
+  def hash128x64(data: Array[Byte], offset: Int, length: Int, seed: Int): (Long, Long) = {
+    // Use an unsigned 32-bit integer as the seed
+    hash128x64Internal(data, offset, length, seed & 0xFFFFFFFFL)
+  }
+
+  /**
+   * Generates 128-bit hash from the byte array with the given offset, length and seed.
+   *
+   * This is an implementation of the 128-bit hash function `MurmurHash3_x64_128`
+   * from Austin Appleby's original MurmurHash3 `c++` code in SMHasher.
+   */
+  private def hash128x64Internal(data: Array[Byte], offset: Int, length: Int, seed: Long): (Long, Long) = {
+    var h1 = seed
+    var h2 = seed
+    val nblocks = length >> 4
+
+    // body
+    for (i <- 0 until nblocks) {
+      val index = offset + (i << 4)
+      var k1 = getLittleEndianLong(data, index)
+      var k2 = getLittleEndianLong(data, index + 8)
+
+      // mix functions for k1
+      k1 *= C1
+      k1 = JLong.rotateLeft(k1, R1)
+      k1 *= C2
+      h1 ^= k1
+      h1 = JLong.rotateLeft(h1, R2)
+      h1 += h2
+      h1 = h1 * M + N1
+
+      // mix functions for k2
+      k2 *= C2
+      k2 = JLong.rotateLeft(k2, R3)
+      k2 *= C1
+      h2 ^= k2
+      h2 = JLong.rotateLeft(h2, R1)
+      h2 += h1
+      h2 = h2 * M + N2
+    }
+
+    // tail
+    var k1 = 0L
+    var k2 = 0L
+    val index = offset + (nblocks << 4)
+
+    @tailrec def switchAllLower(n: Int): Unit = {
```

Comment:
This was a funny bug when ported the Java code. Java's switch falls through the other cases until there is a break.

Reply:
Hehe, I don't remember when but I have definitely hit exactly that in some context.